src/main/java/hudson/plugins/spotinst/repos/LockRepo.java

```diff
@@ -0,0 +1,72 @@
+package hudson.plugins.spotinst.repos;
+
+import hudson.plugins.spotinst.api.SpotinstApi;
+import hudson.plugins.spotinst.api.infra.ApiException;
+import hudson.plugins.spotinst.api.infra.ApiResponse;
+import hudson.plugins.spotinst.api.infra.ExceptionHelper;
+
+public class LockRepo implements ILockRepo {
+
+    @Override
+    public ApiResponse<String> acquireGroupControllerLock(String groupId, String accountId, String controllerIdentifier,
+                                                          Integer ttl) {
+        ApiResponse<String> retVal;
+
+        try {
+            String lockResult = SpotinstApi.AcquireLockGroupController(groupId, accountId, controllerIdentifier, ttl);
+
+            retVal = new ApiResponse<>(lockResult);
+        }
+        catch (ApiException e) {
+            retVal = ExceptionHelper.handleDalException(e);
+        }
+
+        return retVal;
+    }
+
+    @Override
+    public ApiResponse<String> setExpiry(String groupId, String accountId, String controllerIdentifier, Integer ttl) {
+        ApiResponse<String> retVal;
+
+        try {
+            String lockResult = SpotinstApi.ExpandGroupControllerLock(groupId, accountId, controllerIdentifier, ttl);
```

Comment:
rename ExpandGroupControllerLock to setGroupControllerLockExpiry

Reply:
done